xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -797,6 +814,7 @@ static void DisplayReconfigured(CGDirectDisplayID display,
   m_nWidth = newWidth;
   m_nHeight = newHeight;
   m_glContext = context;
+  g_graphicsContext.SetFPS(m_refreshRate);
```

Comment:
i have put a fix in my https://github.com/Memphiz/xbmc/commits/vtb_choppy_bp branch and jenkins does a build right now. I let the users test then.

It contains - the fix for the window origin + your idispresource commit + setting the graphicscontext fps before calling OnResetDisplay callbacks and i've put the wait before the flushbuffers...